### PR TITLE
Give reader role to dry run account for datasets in stage

### PR DESCRIPTION
**Pull request by BenWu**
## Description

This attempts to add the dry run function service account as a reader to datasets created in stage.  Currently, dry run accounts have the reader role at the project level which doesn't match how it's set in shared-prod, resulting in cases where dry runs succeed in stage but fail on main/prod (e.g. querying certain information schema views).  Granting at the dataset level should make it similar to prod so only the job user role is set in the project.

**Reviewer, please follow [this checklist](https://github.com/mozilla/bigquery-etl/.github/reviewer_checklist.md)**

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-5064)

### CODE REVIEW

File: bigquery_etl/cli/stage.py [hunk 1]
```diff
@@ -104,8 +105,8 @@ def deploy(
         # copy SQL to a temporary directory
         tmp_dir = Path(tempfile.mkdtemp())
         tmp_dir.mkdir(parents=True, exist_ok=True)
-        shutil.copytree(sql_dir, tmp_dir)
-        sql_dir = tmp_dir / sql_dir.name
+        shutil.copytree(sql_dir, tmp_dir, dirs_exist_ok=True)
+        sql_dir = tmp_dir / Path(sql_dir).name
```
Comment (BenWu) [inline bigquery_etl/cli/stage.py:109]: The `--copy-sql-to-tmp-dir` wasn't working when I was using it locally

File: bqetl_project.yaml [hunk 2]
```diff
@@ -20,6 +20,9 @@ default:
 
 dry_run:
   function: https://us-central1-moz-fx-data-shared-prod.cloudfunctions.net/bigquery-etl-dryrun
+  function_accounts:
+  - bigquery-etl-dryrun@moz-fx-data-shared-prod.iam.gserviceaccount.com
+  - bigquery-etl-dryrun@moz-fx-data-shar-nonprod-efed.iam.gserviceaccount.com
```
Comment (BenWu) [inline bqetl_project.yaml:25]: It's possible to programatically get the service accounts from the url which is what I did first (in https://github.com/mozilla/bigquery-etl/pull/6288/commits/16fba8a72538e024b971bf4b362620f6d0d3a78a) but it doesn't seem worth the code complexity

**Review by BenWu — COMMENTED** [review]
I made some trivial changes to sql to verify that deploys are still working https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/39780/workflows/21cd9111-f84b-4457-b534-494660ccf020/jobs/454112

**Review by scholtzan — APPROVED** [review]
I guess we'll have to remove project-level permissions for the dry run accounts once this is merged?

**Comment by BenWu** [comment]
Yes the roles can be changed to `bigquery.jobUser` and `bigquery.readSessionUser` to match shred-prod.  I'll wait a bit before changing them to make sure this change by itself doesn't break anything unexpected

**Pull request merged**